### PR TITLE
added bit manipulation

### PR DESCRIPTION
**Pull request by saumyasrivastava08**
I have fixed the issue #3111 
@iamrajiv please review it and merge it.

### CODE REVIEW

File: C-Plus-Plus/Bit--Manipulation/1.1 Check if Kth bit is set or not.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+//check if the kth bit is set or not
```
Comment (iamrajiv) [inline C-Plus-Plus/Bit--Manipulation/1.1 Check if Kth bit is set or not.cpp]: @saumyasrivastava08 Rename this program to `Check_Kth_bit_is_set_or_not.cpp`.

File: C-Plus-Plus/Bit--Manipulation/2.1 Count No of Set Bits.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//Count No of bits
```
Comment (iamrajiv) [inline C-Plus-Plus/Bit--Manipulation/2.1 Count No of Set Bits.cpp]: @saumyasrivastava08 Rename this program to `Count_set_bits.cpp`.

File: C-Plus-Plus/Bit--Manipulation/2.2 Brian Algo.cpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
+//Power of 2
```
Comment (iamrajiv) [inline C-Plus-Plus/Bit--Manipulation/2.2 Brian Algo.cpp]: @saumyasrivastava08 Rename this program to `Brian_Kernighan's_Algorithm.cpp`.

File: C-Plus-Plus/Bit--Manipulation/3.1 Power of 2.cpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
+//Power of 2 (Naive Approach)
```
Comment (iamrajiv) [inline C-Plus-Plus/Bit--Manipulation/3.1 Power of 2.cpp]: @saumyasrivastava08 Rename this program to `Is_power_of_2.cpp`.

File: C-Plus-Plus/Bit--Manipulation/4.1 One odd occuring.cpp [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// Only one odd ocurring number
```
Comment (iamrajiv) [inline C-Plus-Plus/Bit--Manipulation/4.1 One odd occuring.cpp]: @saumyasrivastava08 Rename this program to `One_odd_occuring.cpp`.

File: C-Plus-Plus/Bit--Manipulation/5.1 Two odd occuring elements.cpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
+// Only two odd ocurring number
```
Comment (iamrajiv) [inline C-Plus-Plus/Bit--Manipulation/5.1 Two odd occuring elements.cpp]: @saumyasrivastava08 Rename this program to `Two_odd_occuring.cpp`.

File: C-Plus-Plus/Bit--Manipulation/6.1 Power Set Using Bitwise.cpp [hunk 7]
```diff
@@ -0,0 +1,32 @@
+#include <iostream>
```
Comment (iamrajiv) [inline C-Plus-Plus/Bit--Manipulation/6.1 Power Set Using Bitwise.cpp]: @saumyasrivastava08 Rename this program to `Power_set_using_bitwise.cpp`.

File: C-Plus-Plus/Bit--Manipulation/7.1 Swap even nd odd bits.cpp [hunk 8]
```diff
@@ -0,0 +1,23 @@
+// C++ program to swap even and
```
Comment (iamrajiv) [inline C-Plus-Plus/Bit--Manipulation/7.1 Swap even nd odd bits.cpp]: @saumyasrivastava08 Rename this program to `Power Set Using Bitwise.cpp`.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@saumyasrivastava08 Update the readme file by adding every program link in it. Also, add sample input-output, space and time complexity in every program. I would also suggest adding a little bit of description of the program and problem link in code comments.

**Comment by saumyasrivastava08** [comment]
@iamrajiv check

**Review by iamrajiv — COMMENTED** [review]
@saumyasrivastava08 Add sample input-output in every program. Also, space and time complexity is not added to every program.

**Comment by saumyasrivastava08** [comment]
@iamrajiv done can you review

**Review by iamrajiv — APPROVED** [review]

**Comment by iamrajiv** [comment]
@all-contributors please add @saumyasrivastava08 for her code

**Pull request merged**